### PR TITLE
bkpr-listaccountevents filter by id

### LINKED ISSUE #7518: bkpr - get entries by payment id
When trying to lookup a specific payment with bkpr, I've been using `bkpr-listaccountevents` and jq or grep to filter for a specific payment. That is not very pretty and starting to get slow because `bkpr-listaccountevents` is listing everything it has in its database.

It would be nice to `lightning-cli -k bkpr-listaccountevents payment_id=<id>` so that bkpr only fetches relevant entries from its database.

### PR DESCRIPTION
**Pull request by michael1011**
Closes #7518

It's much nicer to ask the database to filter than getting *everything* and filtering the output.

```
time ./cli/lightning-cli --regtest -k bkpr-listaccountevents | grep x 
         "payment_id": "x",

real	0m6,908s
user	0m1,745s
sys	0m0,534s
```

```
time ./cli/lightning-cli --regtest -k bkpr-listaccountevents payment_id=x 
{
   "events": [
      {
         "payment_id": "x",
         <omitted data>
      }
   ]
}

real	0m0,074s
user	0m0,002s
sys	0m0,000s
```

### CODE REVIEW

File: plugins/bkpr/bookkeeper.c [hunk 1]
```diff
@@ -386,6 +395,16 @@ static struct command_result *json_list_account_events(struct command *cmd,
 		channel_events = account_get_channel_events(cmd, db, acct);
 		chain_events = account_get_chain_events(cmd, db, acct);
 		onchain_fees = account_get_chain_fees(cmd, db, acct);
+	} else if (payment_id != NULL) {
+		channel_events = get_channel_events_by_id(cmd, db, payment_id);
+
+		// Transaction ids are stored as big-endian in the database
```
Comment (niftynei) [inline plugins/bkpr/bookkeeper.c:401]: nit: comments in cln are always done as /* block */ style

File: plugins/bkpr/bookkeeper.c [hunk 2]
```diff
@@ -386,6 +395,16 @@ static struct command_result *json_list_account_events(struct command *cmd,
...
+		reverse_bytes(payment_id->u.u8, sizeof(payment_id->u.u8));
+		tx_id = tal(cmd, struct bitcoin_txid);
+		tx_id->shad.sha = *payment_id;
```
Comment (niftynei) [inline plugins/bkpr/bookkeeper.c:404]: nit: put the payment_id data into the tx_id, and then reverse the bytes. this leaves the `payment_id` data untouched/true to self.

**Review by niftynei — COMMENTED** [review]
ACK bb328772d79783ebf87a2f50f5b079866cb473ec

two tiny nits that would be nice to fixup but largely optional imo. i think the CI error is a global/master one?

**Comment by michael1011** [comment]
Addressed the comments.

> i think the CI error is a global/master one?

It seems to me like CI is little flaky. Let's see if the new commit passes that step

**Pull request merged**